src/services/rest/RestService.ts

```diff
@@ -0,0 +1,148 @@
+import {App} from "../../ManagementConsole";
+import {IRestRequest} from "./IRestRequest";
+import {ISocketBinding} from "../socket-binding/ISocketBinding";
+import {SocketBindingService} from "../socket-binding/SocketBindingService";
+import IHttpPromise = angular.IHttpPromise;
+
+const module: ng.IModule = App.module("managementConsole.services.rest", []);
+
+export class RestService {
+
+  static $inject: string[] = ["$http", "$cacheFactory", "$q", "$location", "socketBindingService"];
+
+  private restBinding: ISocketBinding;
+
+  constructor(private $http: ng.IHttpService,
+              private $cacheFactory: ng.ICacheFactoryService,
+              private $q: ng.IQService,
+              private $location: ng.ILocationService,
+              private socketBindingService: SocketBindingService) {
+
+    // TODO somehow discover installed rest binding better than this?
+    socketBindingService.getDefaultSocketBindingGroup().then((result: ISocketBinding []) => {
+      angular.forEach(result, (value) => {
+        if (value.name === "rest") {
+          this.restBinding = value;
+        }
+      });
+    });
+  }
+
+  executeCacheGet(cacheTarget: string, key: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("get", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheDelete(cacheTarget: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: undefined
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("delete", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheDeleteKey(cacheTarget: string, key: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
```

Comment:
We should add the `/` in generateURL if the suffix is not undefined as currently you are repeating this many times.

Reply:
+1